gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/transformer/RequestPreProcessor.java

```diff
@@ -76,12 +76,20 @@ public EntitiesRequest transform(
         AttributeMetadataUtil.getAttributeIdMappings(
             attributeMetadataProvider, context, originalRequest.getEntityType());
     EntitiesRequest.Builder entitiesRequestBuilder = EntitiesRequest.newBuilder(originalRequest);
+
+    // Convert the time range into a filter and set it on the request so that all downstream
+    // components needn't treat it specially.
+    Filter filter = TimeRangeFilterUtil.addTimeRangeFilter(
```

Comment:
Just thinking out loud:
An alternative approach to this could be to only to modify the ExecutionTreeBuilder.buildFilterTree to take in the EntitiesRequest and construct the DataFetcherNode for the timestamp condition too.
That way all of this preprocessing and and special casing in the QueryServiceEntityFetcher can probably be avoided. i.e. they continue to work with the first class time fields in the EntitiesRequest.

Reply:
@avinashkolluru That's already being done. That was my previous PR :) However, the non-leaf nodes don't take the time range filter because the only filter they apply is based on child node's results. That ended up removing time range filter for metric queries and time series queries, which is what caused another bug, to which this PR is a fix.

Here is the latest status:
* Time range is added to the request filters in the preprocessor
* Execution tree creates a separate DataFetcherNode for this
* QueryServiceEntityFetcher checks if there is time range filter, adds it if not. I'm not really sure this logic is clean so, if possible, I'd like to fix this cleanly in the next iteration.

Reply:
I think i wasn't clear or may there is some disconnect. Here is what i was thinking:
1) Only modify the buildFilterTree to construct and extra DataFetcherNode based on the start/end time in the EntitiesRequest
2) The non leaf nodes i.e. SelectionNode, continues to work how it was earlier, where the EntitiesRequest filter(in ExecutionVisitor.accept()) is only based on its child nodes results. But, internally in QueryServiceDataFetcher, we anyways use the start/end time from the EntitiesRequest and construct the appropriate query-service request. Basically, all of this doesn't change at all.

So, just by doing 1), I think we would have solved the original problem where the initial request to query-service was going without the time range(because the new DataFetcherNode based on the time range will take care of it)

Reply:
Thanks for clarifying and I see the approach now. IMO, hard-coding time range filter in QueryServiceEntityFetcher isn't clean and hence I was thinking of approaches to get rid of that instead. Let me think a bit more on this.